dev/set_matrix.py

```diff
@@ -0,0 +1,427 @@
+"""
+A script to set a matrix for the cross version tests for MLflow Models / autologging integrations.
+
+# How to run:
+
+python dev/set_matrix.py --diff-only \
+    --ref-config <URL or local file path> \
+    --changed-files mlflow/sklearn/__init__.py
+
+# How to run doctests:
+
+pytest dev/set_matrix.py --doctest-modules --verbose
+"""
+
+import argparse
+from distutils.version import LooseVersion
+import json
+import operator
+import os
+import re
+import shutil
+import urllib.request
+
+import yaml
+
+CONFIG_PATH = "ml-package-versions.yml"
+
+
+def read_yaml(location):
+    """
+    Reads a YAML file. `location` can be a URL or local file path.
+    """
+    if re.search("^https?://", location):
+        with urllib.request.urlopen(location) as f:
+            return yaml.load(f, Loader=yaml.SafeLoader)
+    else:
+        with open(location) as f:
+            return yaml.load(f, Loader=yaml.SafeLoader)
+
+
+def get_released_versions(package_name):
+    """
+    Fetches the released versions & datetimes of the specified Python package.
+    """
+    url = "https://pypi.python.org/pypi/{}/json".format(package_name)
+    data = json.load(urllib.request.urlopen(url))
+
+    versions = {
+        ver: files[0]["upload_time"] for ver, files in data["releases"].items() if len(files) > 0
```

Comment:
Is it ever possible that "upload_time" is missing from a release? Can we add a comment explaining why we're selecting the first element of `files`?

Reply:
Added a commen in the `get_released_versions` :)